search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java

```diff
@@ -96,42 +89,48 @@ void handlerReturnsSearchResultsWithEmptyHitsWhenQueryResultIsEmpty() throws IOE
     }
 
     @Test
-    void handlerThrowsExceptionWhenGatewayIsBad() throws IOException {
-        var elasticSearchClient = new SearchClient(setUpBadGateWay());
-        var handler = new SearchResourcesApiHandler(environment, elasticSearchClient);
-        Executable executable = () -> handler.processInput(null, getRequestInfo(), mock(Context.class));
-        assertThrows(ApiGatewayException.class, executable);
+    void shouldReturnProblemResponseWhenGatewayIsBad() throws IOException {
+        prepareRestHighLevelClientBadGateWayResponse();
+
+        handler.handleRequest(getInputStream(), outputStream, mock(Context.class));
+
+        GatewayResponse<Problem> gatewayResponse = GatewayResponse.fromOutputStream(outputStream);
+
+        assertNotNull(gatewayResponse.getHeaders());
+        assertEquals(gatewayResponse.getStatusCode(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
Is this the response we want?

Reply:
nope. probably we want Bad Gateway. fix the test name as well.